### ISSUE TITLE
Color Scheme Manager Problems with UVL Models

### ISSUE THREAD

**Issue by bjthehun** [FeatureIDE/FeatureIDE#1183]
#### Prerequisitives
* **Eclipse Version:** 2021-06 (4.20)
* **FeatureIDE-Version:** 3.7.2
* **Operating system:** Ubuntu 21.04
* **Java runtime used to run eclipse:** OpenJDK 64-Bit Server VM 16.0.1 
* **FeatureIDE Composer:** Feature Modeling

### Issue Description
The Color Scheme Manager for FeatureIDE projects does not work for Feature Models in UVL format. Instead of opening the manager, Eclipse logs the following error:

```
org.eclipse.core.commands.ExecutionException: Error executing 'de.ovgu.featureide.fm.ui.commands.ManageColorSchemes': java.lang.NullPointerException: Cannot invoke "java.nio.file.Path.toAbsolutePath()" because "file" is null
	at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:170)
	at org.eclipse.core.commands.Command.executeWithChecks(Command.java:488)
	at org.eclipse.core.commands.ParameterizedCommand.executeWithChecks(ParameterizedCommand.java:487)
	at org.eclipse.e4.core.commands.internal.HandlerServiceImpl.executeHandler(HandlerServiceImpl.java:213)
	at org.eclipse.e4.ui.workbench.renderers.swt.HandledContributionItem.executeItem(HandledContributionItem.java:438)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.handleWidgetSelection(AbstractContributionItem.java:449)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.lambda$2(AbstractContributionItem.java:475)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:5879)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1427)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:5121)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4599)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1157)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1046)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:644)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:551)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:156)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:152)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:401)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:654)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:591)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1462)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1435)
Caused by: org.eclipse.e4.core.di.InjectionException: java.lang.NullPointerException: Cannot invoke "java.nio.file.Path.toAbsolutePath()" because "file" is null
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:68)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:319)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:253)
	at org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:173)
	at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:156)
	... 33 more
Caused by: java.lang.NullPointerException: Cannot invoke "java.nio.file.Path.toAbsolutePath()" because "file" is null
	at de.ovgu.featureide.fm.core.color.FeatureColorManager.getProject(FeatureColorManager.java:335)
	at de.ovgu.featureide.fm.core.color.FeatureColorManager.getColorSchemes(FeatureColorManager.java:318)
	at de.ovgu.featureide.fm.ui.wizards.ColorSchemePage.<init>(ColorSchemePage.java:66)
	at de.ovgu.featureide.fm.ui.wizards.ColorSchemeWizard.addPages(ColorSchemeWizard.java:47)
	at org.eclipse.jface.wizard.WizardDialog.createContents(WizardDialog.java:632)
	at org.eclipse.jface.window.Window.create(Window.java:431)
	at org.eclipse.jface.dialogs.Dialog.create(Dialog.java:1094)
	at org.eclipse.jface.window.Window.open(Window.java:788)
	at de.ovgu.featureide.ui.handlers.ManageColorSchemesHandler.singleAction(ManageColorSchemesHandler.java:40)
	at de.ovgu.featureide.ui.handlers.base.AFeatureProjectHandler.singleAction(AFeatureProjectHandler.java:49)
	at de.ovgu.featureide.fm.ui.handlers.base.ASelectionHandler.execute(ASelectionHandler.java:53)
	at org.eclipse.ui.internal.handlers.HandlerProxy.execute(HandlerProxy.java:283)
	at org.eclipse.ui.internal.handlers.E4HandlerProxy.execute(E4HandlerProxy.java:97)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:58)
	... 37 more

```

However, using Color Schemes from the `Feature Color` context menu entry in a Feature Diagram Editor still works.

### Steps to Reproduce
1. Right-click on a FeatureIDE project with UVL files only in the Package Explorer.
2. Select `FeatureIDE` > `Manage Color Schemes`.

**Comment by Rahel12342** [comment]
This issue seems to be cause by the FeatureIDE dependency on model.xml files. This issue is not UVL specific, it appears for every project not containing a file named model.xml. 
In order to cleanly fix this issue we require to adapt many classes.

**Comment by skrieter** [comment]
Fixed in 4aa18048712a438e7172a56733c585161aafd939
See #1406 for more information.
